src/v/redpanda/application.cc

```diff
@@ -618,6 +629,64 @@ void application::check_environment() {
     }
 }
 
+/// Here we check for too many consecutive unclean shutdowns/crashes
+/// and abort the startup sequence if the limit exceeds
+/// crash_loop_max_limit until the operator intervenes.
+/// This is to prevent on disk state from piling up in each unclean
+/// run and creating more state to recover for the next run.
+void application::check_for_crash_loop() {
+    auto file_path = config::node().data_directory().path
+                     / crash_loop_tracker_file;
+    uint32_t consecutive_crashes = 0;
+    if (ss::file_exists(file_path.string()).get()) {
+        // Ok to read the entire file, it contains a serialized uint32_t.
+        auto buf = read_fully(file_path).get();
+        try {
+            consecutive_crashes = serde::from_iobuf<uint32_t>(std::move(buf));
+        } catch (const serde::serde_exception&) {
+            // A malformed log file, ignore and reset it later.
+            // We truncate it below.
+            vlog(_log.warn, "Ignorning malformed tracker file {}", file_path);
+        }
+    }
+
+    if (
```

Comment:
I think we need a time element to this condition.  If someone runs redpanda on a system that they hard-power-off every weekend for maintenance (I mean, that's a bad idea but it might happen in some environments), then we would end up calling that a "crash loop" after a few weeks, even though the system is staying up.

Someone doing their own container orchestration might also just be in the (bad) habit of hard-killing things when they want to restart, especially in a test/dev environment where they mainly care about speed.

I think the logic could be that at the start of this function, if the last write time of the trackerfile is more than T seconds old, then just delete the file, i.e. treat the previous startup as a success, even if it was shutdown uncleanly, because it appears to have stayed up for a reasonable amount of time.  A reasonable default for T might be one hour.

Reply:
Ya thats a fair point I think. I captured that metadata in the tracker file rather than relying on the mtime of the file itself. The logic is something like 

if (now() - last_failure_start_time > 1h) reset().